opentimelineio/opentime.py

```diff
@@ -524,6 +520,53 @@ def to_frames(time_obj, fps=None):
     return int(time_obj.value_rescaled_to(fps))
 
 
+def valid_timecode_framerate(func):
+    """Decorator to make sure a valid rate is passed"""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        valid_fps = [
+            1,
+            23.976,
+            23.98,
+            24,
+            25,
+            29.97,
+            30,
+            48,
+            59.94,
+            60
+            ]
+
+        if kwargs.get('rate') is not None:
+            rate = kwargs.get('rate')
+
+        # These feel a bit weak. Should be replaced if rate get's refactored
+        elif isinstance(args[0], RationalTime):
+            rate = args[0].rate
+
+        # These feel a bit weak. Should be replaced if rate get's refactored
+        elif isinstance(args[0], (str)):
+            rate = args[1]
+
+        else:
+            raise ValueError('Unable to find rate in the passed arguments.')
+
+        if not any(r == rate for r in valid_fps):
```

Comment:
I did it this way in case someone passes the integer framerate's as floats. (24.0 vs 24 etc.). 
If we stick to and enforce rate always being a float I can update the VALID_TIMECODE_RATES to all floats and do the `rate in valid_fps` test.